src/Auth0Client.ts

```diff
@@ -731,11 +749,9 @@ export default class Auth0Client {
     if (localOnly) {
       return;
     }
-
-    const federatedQuery = federated ? `&federated` : '';
-    const url = this._url(`/v2/logout?${createQueryParams(logoutOptions)}`);
-
-    window.location.assign(`${url}${federatedQuery}`);
+    delete options.localOnly;
```

Comment:
We should not be doing this as this deletes the localOnly from the options object being passed to the logout, meaning any subsequent call to `logout` will now not have the localOnly property.

```
const options = {
  localOnly = true
};
client.logout(options); 
client.logout(options); // localOnly will not exist anymore
```

Reply:
I didn't realize `delete` had this impact. 

Perhaps I'm confused on your desired outcome. Do you want to pass an object to `buildLogoutUrl` that contains or does not contain the `localOnly` option? 

If it should be passed, are you saying that we should just pull it out in the parsing of options so as to prevent it from being in the `logoutOptions` object, e.g.

```
const { federated, localOnly, ...logoutOptions } = options;
```

Reply:
My desired outcome is to not change any existing functionality by introducing this change to ensure we dont break anything for our users.

I think we should not pass a property `localOnly` to buildLogoutUrl as it has no meaning in that context.

I think we can use this:

```
const { localOnly, ...logoutOptions } = options;
if (localOnly && logoutOptions.federated) {
  ...
}
const url = this.buildLogoutUrl(logoutOptions);
```

Reply:
I see what you're saying. Thank you for the clarification.